egui/src/ui.rs

```diff
@@ -883,6 +883,25 @@ impl Ui {
         self.add(TextEdit::multiline(text))
     }
 
+    /// A `TextEdit` for code editing.
+    ///
+    /// Se also [`TextEdit::code_editor`].
+    pub fn code_editor(&mut self, text: &mut String) -> Response {
+        self.add(TextEdit::multiline(text).code_editor())
+    }
+
+    /// A `TextEdit` for code editing with configurable `Tab` management.
+    ///
+    /// Se also [`TextEdit::code_editor_with_config`].
+    pub fn code_editor_with_config(
+        &mut self,
+        text: &mut String,
+        tab_as_spaces: bool,
+        tab_moves_focus: bool,
+    ) -> Response {
```

Comment:
It is bad form to have multiple boolean arguments, as it leads to code that is very hard to decipher:

``` rust
TextEdit::code_editor_with_config(code, true, false) // what does it mean!?
```

I think it is better in this case to just have the opinionated `code_editor`

Reply:
I crated a `CodingConfig` struct for this purpose.
https://github.com/DrOptix/egui/blob/e14c5491b61574647ed837130e0a61f4c1ce98b4/egui/src/widgets/text_edit.rs#L107-L112